### PR TITLE
Allow Jolt to cure cardiac arrest in people with SMES human

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[FEATURE] [BALANCE]
## About the PR 
[This thread](https://forum.ss13.co/showthread.php?tid=18096) pointed out that arcfiends are often practically dead if they ever enter cardiac arrest.
This PR makes Jolt work differently on people with SMES human (like arcfiends):
if cast on someone else there is a 20% chance per cycle to cure cardiac arrest and heart failure, with 100% chance to cure at the end of the channel
if cast on yourself it immediately cures cardiac arrest, deals +30 burn and -100 oxy damage and stuns you for 5 seconds
it can now be self-cast when incapacitated as otherwise there's not much chance you'd ever be able to use it while in arrest

Note that to let Jolt self-cast I had to change targetable/castcheck to include a target field, not really sure why it didn't have this before and it doesn't seem to have broken anything to add it.

## Why's this needed? 
Having little to no realistic way to cure cardiac arrest is sad, and this solution is more interesting than just letting arcfiends be defibbed. (Also allows them to revive each other in the rare situation of two arcfiends working together)
500 energy, a 2 minute cooldown, 30 burn damage, a stun and a loud zappy noise seems balanced enough for a self defib.
A side effect of this is that arcfiends can now accidentally self-cast Jolt and stun themselves.

## Changelog

```changelog
(u)LeahTheTech
(+)Arcfiend's Jolt ability will now cure cardiac arrest in people with SMES human and can now be self cast.
```

### CODE REVIEW

File: code/modules/antagonists/arcfiend/arcfiend.dm [hunk 1]
```diff
@@ -361,13 +361,26 @@ ABSTRACT_TYPE(/datum/targetable/arcfiend)
 
 	cast(atom/target)
 		. = ..()
-		if (target == holder.owner) return TRUE
 		if (!IN_RANGE(holder.owner, target, 1)) return TRUE
 		if (ishuman(target))
+			if (target == holder.owner)
+				self_cast(target)
```
Comment (Sovexe) [inline code/modules/antagonists/arcfiend/arcfiend.dm:367]: I'd rather they not be able to self cast on themselves without reason

File: code/modules/antagonists/arcfiend/arcfiend.dm [hunk 2]
```diff
@@ -361,13 +361,26 @@ ABSTRACT_TYPE(/datum/targetable/arcfiend)
...
+	proc/self_cast(mob/living/carbon/human/self)
+		if (self.find_ailment_by_type(/datum/ailment/malady/flatline))
+			boutput(self, "<span class='alert'>You feel your heart jolt back into life!</span>")
+		else
+			boutput(self, "<span class='alert'>You feel a powerful jolt course through you!</span>")
+		playsound(self, "sound/effects/elec_bigzap.ogg", 30, 1)
+		self.cure_disease_by_path(/datum/ailment/malady/flatline)
+		self.TakeDamage("chest", 0, 30, 0, DAMAGE_BURN)
+		self.take_oxygen_deprivation(-100)
+		self.changeStatus("paralysis", 5 SECONDS)
```
Comment (Sovexe) [inline code/modules/antagonists/arcfiend/arcfiend.dm:382]: this should be followed up with a `force_laydown_standup` call

**Review by Sovexe — CHANGES_REQUESTED** [review]
.

File: code/modules/antagonists/arcfiend/arcfiend.dm [hunk 3]
```diff
@@ -361,13 +361,26 @@ ABSTRACT_TYPE(/datum/targetable/arcfiend)
...
 			actions.start(new/datum/action/bar/private/icon/jolt(holder.owner, target, holder, wattage), holder.owner)
 			logTheThing("combat", holder.owner, target, "[key_name(holder.owner)] used <b>[src.name]</b> on [key_name(target)] [log_loc(holder.owner)].")
 		else return TRUE
 
+	proc/self_cast(mob/living/carbon/human/self)
+		if (self.find_ailment_by_type(/datum/ailment/malady/flatline))
+			boutput(self, "<span class='alert'>You feel your heart jolt back into life!</span>")
+		else
+			boutput(self, "<span class='alert'>You feel a powerful jolt course through you!</span>")
+		playsound(self, "sound/effects/elec_bigzap.ogg", 30, 1)
```
Comment (Sovexe) [inline code/modules/antagonists/arcfiend/arcfiend.dm:378]:
```suggestion
		playsound(self, 'sound/effects/elec_bigzap.ogg', 30, 1)
```
no need for sound cache, you arent string building

Comment (TobleroneSwordfish) [reply code/modules/antagonists/arcfiend/arcfiend.dm:378]: oh I did not know about that caching behaviour, I just copied from a different arcfiend ability

**Pull request merged**